File: 344/Easy.jl
# https://www.reddit.com/r/dailyprogrammer/comments/7j33iv/20171211_challenge_344_easy_baumsweet_sequence/

function baumsweet(n)
  n == 0 && return 1
  m = n
  while m & 3 == 0
    m = m >> 2
  end
  m & 1 == 0 ? 0 : baumsweet((m-1)>>1)
end

function baumsweetSeq(n)
  [baumsweet(i) for i in 1:n]
end

File: 381/Easy.hs
-- https://www.reddit.com/r/dailyprogrammer/comments/dv0231/20191111_challenge_381_easy_yahtzee_upper_section/

module Main where

import qualified Data.Map.Strict as M
import           Data.List (foldl')

yahtzee_upper :: [Int] -> Int
yahtzee_upper = maxValue . foldl' (\m x -> M.insertWith (+) x x m) M.empty

maxValue :: (Ord a, Num a) => M.Map k a -> a
maxValue = M.foldl' max 0

testLargeInput :: IO Int
testLargeInput = readFile "inp-easy.txt" >>= return . yahtzee_upper . map read . lines

main :: IO ()
main = testLargeInput >>= putStrLn . show
